lib/piezo.js

```diff
@@ -247,12 +247,34 @@ Piezo.prototype.frequency = function(frequency, duration) {
   return this.tone(tone, duration);
 };
 
+Piezo.prototype.stringToSong = function(stringSong, beats) {
+  beats = beats || 1;
+  var notes = stringSong.match(/\S+/g);
+  var song = [];
+  while (notes.length) {
+    var note = notes.shift();
+    if (/^[0-9]+$/.test(note)) {
+      note = parseInt(note);
+    }
+    var lastNote = song[song.length - 1];
+    if (lastNote && lastNote[0] === note) {
+      lastNote[1] += beats;
+    } else {
+      song.push([note, beats]);
+    }
+  }
+  return song;
+};
```

Comment:
Maybe hoist up those `var` declarations (`note`, `lastNote`)  to the top of the function for clarity as we hurtle toward the future of ES6 block-level scope?